### ISSUE TITLE
Unsupported major.minor version 52.0 error

### ISSUE THREAD

**Issue by teamwork523** [Fallwizard/Fallwizard#6]
I am working on a Dropwizard framework based web application, but I really love the Spring Dependency Injection feature. Therefore, I found Fallwizard, and gave it a try. However, after switching from "io.dropwizard.Configuration" into "io.github.fallwizard.configuration.FallwizardConfiguration". I ran into the following Exception when I started my application:

```
Exception in thread "main" java.lang.UnsupportedClassVersionError: io/github/fallwizard/configuration/FallwizardConfiguration : Unsupported major.minor version 52.0
    at java.lang.ClassLoader.defineClass1(Native Method)
    at java.lang.ClassLoader.defineClass(ClassLoader.java:800)
    at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
    at java.net.URLClassLoader.defineClass(URLClassLoader.java:449)
    at java.net.URLClassLoader.access$100(URLClassLoader.java:71)
    at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
    at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
    at java.lang.ClassLoader.defineClass1(Native Method)
    at java.lang.ClassLoader.defineClass(ClassLoader.java:800)
    at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
    at java.net.URLClassLoader.defineClass(URLClassLoader.java:449)
    at java.net.URLClassLoader.access$100(URLClassLoader.java:71)
    at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
    at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
    at java.lang.Class.getDeclaredMethods0(Native Method)
    at java.lang.Class.privateGetDeclaredMethods(Class.java:2615)
    at java.lang.Class.getMethod0(Class.java:2856)
    at java.lang.Class.getMethod(Class.java:1668)
    at sun.launcher.LauncherHelper.getMainMethod(LauncherHelper.java:494)
    at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:486)
```

The java version on my local box is 1.7.0_76. One of the [StackOverflow thread](http://stackoverflow.com/questions/10382929/how-to-fix-unsupported-major-minor-version-51-0-error) mentioned about the similar issue. I noticed that the version 52.0 indicates JDK 1.8, so my question is does Fallwizard requires Java 8?

**Comment by jar349** [comment]
So, I've spent some time this morning looking into this problem.  It's true that the error you got indicates a java version mismatch problem.  There are generally 3 places that we could have specified the Java version: 
1) in the build.gradle properties file
2) spring IO platform versions might have specified the java version
3) manually added in jar!:/META_INF/MANIFEST.MF

I checked all 3 of these places and in none of them do we specify the version.  Moreover, the project page for Spring IO Platform says that it supports both 1.7 and 1.8.  So the only thing I can think of at this point is that the bytecode got generated in a 1.8-specific way that is not compatible with 1.7 (remember that 1.8 introducted lambdas - who knows what changes that required to bytecode?).

I'll check to make sure we're not using any 1.8-specific features and, if not, I can upload a new version of Fallwizard compiled using a 1.7 JDK.  Give me a day or two.

In the meantime, if you just need to make progress, you can compile the JAR yourself from source code and include it as a local dependency until the new version is available on Maven Central.

**Comment by jar349** [comment]
FWIW, I took a look at the bytecode and confirmed that it targets 52.0 (which we already knew).  So yes, I'll compile targeting an earlier JDK (assuming we're not using 1.8-specific stuff).

```
$ javap -verbose FallwizardConfiguration.class 
Classfile /storage/downloads/fw/io/github/fallwizard/configuration/FallwizardConfiguration.class
  Last modified Feb 3, 2015; size 889 bytes
  MD5 checksum 28d4891c4e7445081454654ca5f10258
  Compiled from "FallwizardConfiguration.java"
public class io.github.fallwizard.configuration.FallwizardConfiguration extends io.dropwizard.Configuration
  minor version: 0
  major version: 52
  flags: ACC_PUBLIC, ACC_SUPER
```

**Comment by teamwork523** [comment]
@jar349 Thank you so much for looking into this issue. I am glad you find the root cause, and looking forward to see a 1.7 JDK compatible version of Fallwizard. I will go with the local build version in the meantime. Thanks.

**Comment by jar349** [comment]
I've uploaded version 0.8.1.0 to maven central.  It could take up to two hours for it to be visible.  I compiled the jar using JDK 1.8, but I've specified the source and target compatibility to be 1.7.  I confirmed that the .class files reference version 51.0 rather than 52.0.  So this should work for you.

Additionally, 0.8.1.0 now uses the latest dropwizard (0.8.1) and the latest spring io platform (1.1.2).
